joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/astcreation/AstCreator.scala

```diff
@@ -244,7 +246,35 @@ class AstCreator(
 
   private[astcreation] def tryWithSafeStackOverflow[T](expr: => T): Try[T] = {
     try {
-      Try(expr)
+
+      /** JavaParser throws UnsolvedSymbolExceptions if a type cannot be solved, which is usually an expected occurrence
+        * that does not warrant specific failure logging. Since it's impossible to tell whether these are legitimately
+        * unresolved types or a bug, don't log them.
+        */
+      Try(expr) match {
+        case success: Success[_]                         => success
+        case Failure(exception: UnsolvedSymbolException) => Failure(exception)
+        case failure: Failure[_] =>
+          val exceptionType = failure.exception.getClass
+
+          val shouldLog = loggedExceptionCounts.get(exceptionType) match {
```

Comment:
You will need to use an atomic read + update operation on the concurrent map. Like this the code is not thread safe.